app/javascript/packs/pages/provider_signup/SignupStepper.jsx

```diff
@@ -137,7 +137,17 @@ export default function SignupStepper({ steps }) {
   const CurrentStep = steps[activeStep] && steps[activeStep].component;
 
   if (redirectId) {
-    return <Redirect to={`/providers/${redirectId}?success`} />
+    return (
+      <Redirect
+        push
+        to={{
+          pathname: `/providers/${redirectId}`,
+          state: {
+            providerCreation: true,
+          },
+        }}
+      />
+    )
```

Comment:
Please note that this method adds this attribute to the browser state.

The user will either have to replace the state (open a new window, navigate to new URL) in order for the state to clear.

Other users will not be impacted.

Reply:
Awesome! I didn't know this existed